rq/worker.py

```diff
@@ -560,6 +560,8 @@ def prepare_job_execution(self, job):
         """
         timeout = (job.timeout or 180) + 60
 
+        push_connection(self.connection)
```

Comment:
One more minor thing. Should we move this statement to `perform_job()`, right before `prepare_job_execution()` is called? Having `push_connection()` and `pop_connection()` in the same scope makes more sense.